install/kubernetes/ingress-sni-zvpn.yaml

```diff
@@ -0,0 +1,364 @@
+# Example configuration using Envoy v2 bootstrap, custom envoy configuration with SNI and
+# sample 'zero vpn' port forwarding. This needs to be manually updated for new domains and services,
+# work in progress to automate it by having Pilot generate LDS v2 with SNI and capture the zvpn ports from
+# annotations.
+#
+# Setting up SNI in LDSv2:
+# https://www.envoyproxy.io/docs/envoy/latest/faq/sni.html#faq-how-to-setup-sni
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: envoy-config
+  namespace: istio-system
+data:
+  envoy.yaml: |
+    # You can include a custom bootstrap envoy yaml, updating the path in the 'args'
+    # Ingress configuration using Envoy v2 bootstrap.
+    #
+    # Has 'zero vpn' port forwarding for the control plane - allowing machines outside the VPN with proper certificates
+    # to connect to Istio.
+    #
+    # Current config has a sample port open, exposing an internal service. This will be rolled into LDS and automatically
+    # exposed using service annotations in a future version.
+    #
+    # To customize the envoy config, create a config map and update the ingress config to use the custom envoy bootstrap
+    # from the config map.
+    #
+    # Setting up SNI in LDSv2:
+    # https://www.envoyproxy.io/docs/envoy/latest/faq/sni.html#faq-how-to-setup-sni
+    stats_config:
+      # Without this option it create a lot of regex, may run out of space.
+      use_all_default_tags: false
+
+    admin:
+      access_log_path: /dev/stdout
+      address:
+        # Only allow debugging from localhost
+        socket_address: { address: 127.0.0.1, port_value: 15000 }
+
+    dynamic_resources:
+      lds_config:
+        api_config_source:
+          api_type: REST_LEGACY
+          refresh_delay: 10s
+          # Pilot cluster is named 'rds' for compatibility with existing LDS, which hardcodes this name for the RDS API.
+          cluster_names: [rds]
+      cds_config:
+         api_config_source:
+            api_type: REST_LEGACY
+            refresh_delay: 10s
```

Comment:
Why don’t you set this to 1s - same as what we use today.

Reply:
This - as well as tracing, statsd - will be added with the templating option. 

Re. LDS on ingress - it is needed for zvpn, will create all the TCP proxies.